lang/cs/Org.Apache.REEF.Common/metrics/Api/MetricsTag.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+namespace Org.Apache.REEF.Common.Metrics.Api
+{
+    /// <summary>
+    /// Metrics tag class. Tags can be used to group records together at source, sink or other 
+    /// appropriate places. They can also be used in MetricsFilter to filter records at higher level.
+    /// </summary>
+    public sealed class MetricsTag : IMetricTagCommonFunctions
+    {
+        private readonly IMetricsInfo _info;
+        private readonly string _value;
+        
+        /// <summary>
+        /// Constructor for tags.
+        /// </summary>
+        /// <param name="info">Meta data for tags.</param>
+        /// <param name="value">Value of the tag.</param>
+        public MetricsTag(IMetricsInfo info, string value)
```

Comment:
Wouldn't this always be returned by some other method? Hence, it could be `internal`?